sensor/kubernetes/certrefresh/certificates/requester.go

```diff
@@ -187,37 +137,36 @@ func (r *genericRequester[ReqT, RespT]) RequestCertificates(ctx context.Context)
 		}
 	}
 
+	if r.ongoingRequest.Load() {
+		return nil, errors.New("concurrent requests are not supported.")
+	}
+	r.ongoingRequest.Store(true)
+	defer func() {
+		r.ongoingRequest.Store(false)
+	}()
```

Comment:
Why not simply:
```suggestion
	defer r.ongoingRequest.Store(false)
```
?

Reply:
Now there are two functions in the block. 😄